src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -631,18 +681,10 @@ private string PrivateAbsolutePath
         {
             get
             {
-                UriInfo info = EnsureUriInfo();
-                if ((object?)info.MoreInfo == null)
-                {
-                    info.MoreInfo = new MoreInfo();
-                }
-                string? result = info.MoreInfo.Path;
-                if ((object?)result == null)
-                {
-                    result = GetParts(UriComponents.Path | UriComponents.KeepDelimiter, UriFormat.UriEscaped);
-                    info.MoreInfo.Path = result;
-                }
-                return result;
+                Debug.Assert(IsAbsoluteUri);
+
+                MoreInfo info = EnsureUriInfo().MoreInfo;
+                return info.Path ??= GetParts(UriComponents.Path | UriComponents.KeepDelimiter, UriFormat.UriEscaped);
```

Comment:
This doesn't need to be interlocked because it's idempotent?  And I guess we're relying on the runtime's implicit write barrier for reference writes?

Reply:
Exactly, GetParts is idempotent